src/test/test_runner_host.py

```diff
@@ -304,6 +304,205 @@ def test_docker_basic_run(self):
+    def test_get_container_kwargs(self):
+        step = Box(
+            {
+                "uses": "popperized/bin/sh@master",
+                "args": ["ls"],
+                "id": "one",
+                "dir": "/tmp/",
+            },
+            default_box=True,
+        )
+
+        config_dict = {
+            "engine": {
+                "name": "podman",
+                "options": {
+                    "privileged": True,
+                    "hostname": "popper.local",
+                    "domainname": "www.example.org",
+                    "volumes": ["/path/in/host:/path/in/container"],
+                    "environment": {"FOO": "bar"},
+                },
+            },
+        }
+
+        config = ConfigLoader.load(
+            config_file=config_dict, workspace_dir="/path/to/workdir"
+        )
+
+        with PodmanRunner(init_podman_client=False, config=config) as pr:
+            args = pr._get_container_kwargs(step, "alpine:3.9", "container_a")
+
+            self.assertEqual(
+                args,
+                {
+                    "image": "alpine:3.9",
+                    "command": ["ls"],
+                    "name": "container_a",
+                    "volumes": [
+                        "/path/to/workdir:/workspace",
+                        "/path/in/host:/path/in/container",
+                    ],
+                    "working_dir": "/tmp/",
+                    "environment": {"FOO": "bar"},
+                    "entrypoint": None,
+                    "detach": True,
+                    "stdin_open": False,
+                    "tty": False,
+                    "privileged": True,
+                    "hostname": "popper.local",
+                    "domainname": "www.example.org",
+                },
+            )
+
+        # check container kwargs when pty is enabled
+        config = ConfigLoader.load(
+            config_file=config_dict, workspace_dir="/path/to/workdir", pty=True
+        )
+
+        with PodmanRunner(init_podman_client=False, config=config) as pr:
+            args = pr._get_container_kwargs(step, "alpine:3.9", "container_a")
+
+            self.assertEqual(
+                args,
+                {
+                    "image": "alpine:3.9",
+                    "command": ["ls"],
+                    "name": "container_a",
+                    "volumes": [
+                        "/path/to/workdir:/workspace",
+                        "/path/in/host:/path/in/container",
+                    ],
+                    "working_dir": "/tmp/",
+                    "environment": {"FOO": "bar"},
+                    "entrypoint": None,
+                    "detach": False,
+                    "stdin_open": True,
+                    "tty": True,
+                    "privileged": True,
+                    "hostname": "popper.local",
+                    "domainname": "www.example.org",
+                },
+            )
```

Comment:
I didn't know how to implement this,I think I posted this on Slack. This particular end-to-end workflow execution doesn't reference docker in anyway so I really did not know how to implement this for Podman. I'd push some changes regarding this for you to see if I have perfectly understood this

Reply:
this can be the same test as for the docker, but modifying [this line](https://github.com/getpopper/popper/blob/1d7daf2/src/test/test_runner_host.py#L274) so that the `engine_name="podman"` is passed. So copy-pasting that test function, and modifying this argument

Reply:
Thanks for this. I've made some changes which have helped me replicate the error you had when running a workflow. Let me get on fixing that